### PR TITLE
Fix backward motion for graceful controller

### PR DESCRIPTION
**Pull request by ajtudela**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   |  |
| Primary OS tested on | Ubuntu 22.04 |
| Robotic platform tested on | Gazebo classic |
| Does this PR contain AI generated software? | No |

---

## Description of contribution in a few bullet points

- Fix computation of linear and angular velocities to allow correct backward motion.
- It can be backported to other branches.

## Description of documentation updates required from your changes

---

#### For Maintainers: 
- [ ] Check that any new parameters added are updated in docs.nav2.org
- [ ] Check that any significant change is added to the migration guide
- [ ] Check that any new features **OR** changes to existing behaviors are reflected in the tuning guide
- [ ] Check that any new functions have Doxygen added
- [ ] Check that any new features have test coverage
- [ ] Check that any new plugins is added to the plugins page
- [ ] If BT Node, Additionally: add to BT's XML index of nodes for groot, BT package's readme table, and BT library lists

### CODE REVIEW

**Review by SteveMacenski — APPROVED** [review]
@mikeferguson can you please review this as well? I want to make sure I'm not missing anything subtle. This is with respect to https://github.com/open-navigation/opennav_docking/issues/47

File: nav2_graceful_controller/src/smooth_control_law.cpp [hunk 1]
```diff
@@ -59,13 +58,15 @@ geometry_msgs::msg::Twist SmoothControlLaw::calculateRegularVelocity(
...
+  // Invert the curvature if the robot is moving backwards
+  curvature = backward ? -curvature : curvature;
 
   // Adjust the linear velocity as a function of the path curvature to
   // slowdown the controller as it approaches its target
   double v = v_linear_max_ / (1.0 + beta_ * std::pow(fabs(curvature), lambda_));
 
   // Slowdown when the robot is near the target to remove singularity
-  v = std::min(v_linear_max_ * (ego_coords.r / slowdown_radius_), v);
+  v = std::min(v_linear_max_ * fabs(ego_coords.r / slowdown_radius_), v);
```
Comment (SteveMacenski) [inline nav2_graceful_controller/src/smooth_control_law.cpp:69]: isn't `r` always positive now? Shouldn't `(ego_coords.r / slowdown_radius_)` always be positive?

**Review by SteveMacenski — CHANGES_REQUESTED** [review]

File: nav2_graceful_controller/src/smooth_control_law.cpp (hunk 1 above)
Comment (ajtudela) [reply nav2_graceful_controller/src/smooth_control_law.cpp:69]: yeap, that is not necessary anymore.

**Comment by ajtudela** [comment]
The CI failed but its not related to this PR.

**Review by SteveMacenski — APPROVED** [review]

**Comment by HappySamuel** [comment]
Hi

Will this PR being merged into humble version as well?
As i am currently still using humble version, but found that humble version not yet being updated with the latest fix.

**Comment by SteveMacenski** [comment]
I believe it already is in https://github.com/open-navigation/opennav_docking -- but if that's wrong, please open a PR and happy to merge!

**Pull request merged**